### PR TITLE
Subnets and smart proxies api changes

### PR DESCRIPTION
**Pull request by tstrachota**

### CODE REVIEW

File: app/controllers/api/v1/smart_proxies_controller.rb [hunk 1]
```diff
@@ -2,13 +2,15 @@ module Api
...
+      before_filter :check_feature_type, :only => :index
 
       api :GET, "/smart_proxies/", "List all smart_proxies."
+      param :type, String, :desc => "filter by type"
       param :page, String, :desc => "paginate results"
       param :per_page, String, :desc => "number of entries per request"
 
       def index
-        @smart_proxies = SmartProxy.includes(:features).paginate(paginate_options)
+        @smart_proxies = proxies_by_type(params[:type]).includes(:features).paginate(paginate_options)
```
Comment (ohadlevy) [inline app/controllers/api/v1/smart_proxies_controller.rb]: you can use a scope instead

e.g. dns_proxies.paginate...

see https://github.com/theforeman/foreman/blob/develop/app/models/smart_proxy.rb#L23

Comment (tstrachota) [inline app/controllers/api/v1/smart_proxies_controller.rb]: I actually use them but it's wrapped in the function
https://github.com/theforeman/foreman/pull/259/files#L0R52

Is it an issue? Would you prefer to see it unwrapped inplace?

File: app/controllers/api/v1/smart_proxies_controller.rb [hunk 2]
```diff
@@ -46,6 +48,22 @@ def destroy
         process_response @smart_proxy.destroy
       end
 
+      private
+      def proxies_by_type(type)
+        return SmartProxy.try(type.downcase+"_proxies") if not type.nil?
```
Comment (ohadlevy) [inline app/controllers/api/v1/smart_proxies_controller.rb]: what if type is not nil but not an allowed type?

Comment (ohadlevy) [inline app/controllers/api/v1/smart_proxies_controller.rb]: ah, forget that :) i see below

**Comment by tstrachota** [comment]
Tests for filtering smart proxies by type added.

**Comment by ohadlevy** [comment]
merged - thanks @tstrachota

**Pull request closed without merging**